keps/sig-ui/5008-kep-template/README.md

```diff
@@ -0,0 +1,949 @@
+* A more modern UI developer environment
+* A more feature-rich UI
+* A desktop user experience
+* An integration point for other CNCF and external projects, especially those
+  without a UI focus
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+This proposal does not propose to force people who are using the current
+Dashboard to move to Headlamp.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The proposal is to ship the Headlamp desktop experience as a new Kubernetes
+Desktop and the Headlamp in-cluster UI to eventually become the next generation
+of the Kubernetes Dashboard.
+
+Effectively this entails moving the Headlamp project under the Kubernetes
+GitHub organization and under the auspices of Kubernetes SIG UI. To be
+specific, we would propose that the Headlamp project be moved to a repo under
+the Kubernetes project; something to the effect of kubernetes/ui.
+The current repo kubernetes/dashboard would remain but include a note about the
+existence of the other repo. This would allow for users to compare the projects
+and make an informed decision about which works best for their use case. Over
+time, we can measure adoption trends and make decisions based on that data.
+
+The Headlamp branding would be phased out.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+One of Headlamp's strengths is that it was designed to both deliver a great
+end-user experience and be a platform for others to build upon and integrate
+with. This is demonstrated by the wide range of user stories it enables and one
+of the reasons we believe it can become a hub for the community.
+
+_**Note:** Using “Headlamp” in the following examples to avoid confusion._
+
+#### Story 1
+
+As an end user I want to get an overview of my cluster. I start the Headlamp
+Desktop app which picks up my clusters from my kubeconfig and I’m now able to
+explore my clusters.
+
+#### Story 2
+
+As a new Kubernetes user I want to learn more about Kubernetes. I use the
+Headlamp Desktop app to guide me in installing a local kubernetes cluster using
+minikube. Once my local cluster is installed I’m presented with a tutorial
+plugin which guides me through the various aspects of the UI and Kubernetes
+concepts.
+
+_**Note:** We have the minikube installer plugin in POC but the tutorial plugin
+is not something we currently have._
```

Comment:
I'd only mention shipped features here and introduce a Roadmap section that will cover future work and planned features.

Reply:
I think the goal here is to highlight the possibility of having local clusters provisioned due to the plugin system supporting (desktop only) running local tools, which is something that we already support when enabled. The minikube part is an example of this, and we happen to be working on that, but it's not yet complete, so that's mentioned here too.